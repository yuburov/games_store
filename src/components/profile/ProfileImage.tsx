import React, { useState } from "react";
import ReactFileReader from "react-file-reader";
import style from "./profileImage.module.scss";
import mainStyle from "../../styles/main.module.scss";
import { IUserProfile } from "../../types/UserProfile";

interface IProfileImage extends IUserProfile {}

function ProfileImage({ avatar }: IProfileImage): ReturnType<React.FC<IProfileImage>> {
  const defaultAvatarUrl = "https://winnote.ru/wp-content/uploads/2016/01/1454222417_del_recent_avatar1.png";
  const isImg = avatar || defaultAvatarUrl;
  const [image, setImage] = useState<string>(isImg);

  const handleFiles = (files: { base64: React.SetStateAction<string> }) => {
    setImage(files.base64);
  };

  return (
    <div>
      <img className={style.image} src={image} alt="" />
      <ReactFileReader base64 handleFiles={handleFiles}>
        <button className={mainStyle.btn} type="button">
          Change profile image
        </button>
      </ReactFileReader>
    </div>
  );
}

export default ProfileImage;
