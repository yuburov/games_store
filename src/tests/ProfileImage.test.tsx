import renderer from "react-test-renderer";
import ProfileImage from "../components/profile/ProfileImage";

describe("ProfileInfo component tests", () => {
  test("should render the default avatar", () => {
    const component = renderer.create(<ProfileImage />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
