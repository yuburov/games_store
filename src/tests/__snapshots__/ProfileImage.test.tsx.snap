// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileInfo component tests should render the default avatar 1`] = `
<div>
  <img
    alt=""
    src="https://winnote.ru/wp-content/uploads/2016/01/1454222417_del_recent_avatar1.png"
  />
  <div
    className="react-file-reader"
  >
    <input
      accept="image/*"
      className="react-file-reader-input"
      disabled={false}
      id="844e7363-e923-404d-802a-bdc8a768dd8f"
      multiple={false}
      onChange={[Function]}
      style={
        {
          "opacity": "0",
          "position": "fixed",
          "width": "0px",
        }
      }
      type="file"
    />
    <div
      className="react-file-reader-button"
      onClick={[Function]}
    >
      <button
        type="button"
      >
        Change profile image
      </button>
    </div>
  </div>
</div>
`;
